libs/langchain/langchain/chat_models/huggingface_llama2.py

```diff
@@ -0,0 +1,162 @@
+from typing import Any, Dict, List, Optional, Union
+
+import torch
+from transformers import StoppingCriteria, StoppingCriteriaList
+from transformers.pipelines import TextGenerationPipeline
+
+from langchain.callbacks.manager import (
+    CallbackManagerForLLMRun,
+)
+from langchain.chat_models.base import BaseChatModel
+from langchain.pydantic_v1 import root_validator
+from langchain.schema import ChatResult
+from langchain.schema.messages import (
+    AIMessage,
+    BaseMessage,
+    HumanMessage,
+    SystemMessage,
+)
+from langchain.schema.output import ChatGeneration
+
+B_INST, E_INST = "[INST]", "[/INST]"
+B_SYS, E_SYS = "<<SYS>>", "<</SYS>>"
+
+
+class StoppingCriteriaSub(StoppingCriteria):
```

Comment:
This class is bit lacking in documentation - it is not obvious how to use it, it should be tested and have a bit better typing/docstrings.